File: con-modulos/DoubleDispatch/model/Game.js
import { Board } from './Board.js'
import { Turn } from './Turn.js'

class Game {
  #board;

  #turn;

  constructor() {
    this.#board = new Board();
    this.#turn = new Turn(this.#board);
  }

  getBoard() {
    return this.#board;
  }

  getTurn() {
    return this.#turn;
  }

  getCurrentPlayer() {
    return this.#turn.getCurrentPlayer();
  }

  changeTurn() {
    this.#turn.changeTurn();
  }
  isWinner() {
    return this.#board.isWinner();
  }

  isFinished() {
    return this.#board.isWinner() || this.#board.isComplete();
  }
}

export { Game };

File: designPatterns/controllers/src/main/es/connect4/models/Game.java
package main.es.connect4.models;

import main.es.connect4.types.Color;
import main.es.connect4.types.Coordinate;

public class Game {

    private Board board;
    private Turn turn;

    public Game() {
        this.board = new Board();
        this.turn = new Turn(this.board);
    }

    public void reset() {
        this.board.reset();
        this.turn.resetPlayers();
    }

    public void resetTurn() {
        this.turn.reset();
    }

    public Board getBoard() {
        return board;
    }

    public Turn getTurn() {
        return turn;
    }

    public Color getColor(Coordinate coordinate) {
        return this.board.getColor(coordinate);
    }

    public boolean isGameFinished() {
        return this.board.isGameFinished();
    }

    public boolean isWinner() {
        return this.board.isWinner();
    }

    public int getNumberPlayers() {
        return this.turn.getNumberPlayers();
    }

    public void addPlayer(Player player) {
        this.turn.addPlayer(player);
    }

    public Player getActivePlayer() {
        return this.turn.getActivePlayer();
    }

    public void  play(int column) {
        this.turn.play(column);
    }

}
